File: API/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Link.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({"rel", "class", "href"})
public class Link {
    public String[] rel;
    @JsonProperty("class")
    public String[] klass;
    public String href;

    public Link(String[] rel, String[] klass, String href) {
        this.rel = rel;
        this.klass = klass;
        this.href = href;
    }

    public String[] getRel() {
        return rel;
    }

    public String getHref() {
        return href;
    }
}

File: Android/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/siren/subentities/Link.kt
package ps.leic.isel.pt.gis.hypermedia.siren.subentities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class Link(var rel: Array<String>, @field:JsonProperty("class") var klass: Array<String>?,
           var href: String)
